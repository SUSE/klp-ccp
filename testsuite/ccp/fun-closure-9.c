#include "fun-closure-9.h"

void pu_f(void)
{
	ee_g();
	ee_i();
	ee_j();
}
