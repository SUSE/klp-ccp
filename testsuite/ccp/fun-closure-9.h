/*
 * The toy policy prefers not to externalize function definitions from
 * includeable headers, if possible.
 */

void* ee_g(void);

void *ee_j(void)
{
	return &ee_g;
}

void *ee_i(void)
{
	return &ee_j;
}

void *ee_h(void)
{
	return &ee_i;
}

void *ee_g(void)
{
	return &ee_h;
}
