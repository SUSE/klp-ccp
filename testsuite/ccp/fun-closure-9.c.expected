/* klp-ccp: from testsuite/ccp/fun-closure-9.h */
static void* ee_g(void);

void *ee_j(void);

static void *ee_i(void)
{
	return &ee_j;
}

void *ee_h(void);

void *ee_g(void)
{
	return &ee_h;
}

/* klp-ccp: from testsuite/ccp/fun-closure-9.c */
void klpp_pu_f(void)
{
	ee_g();
	ee_i();
	ee_j();
}
